server/src/main/scala/com/cloudera/livy/LivyConf.scala

```diff
@@ -245,15 +245,25 @@ class LivyConf(loadDefaults: Boolean) extends ClientConf[LivyConf](null) {
   /** Return the spark deploy mode Livy sessions should use. */
   def sparkDeployMode(): Option[String] = Option(get(LIVY_SPARK_DEPLOY_MODE)).filterNot(_.isEmpty)
 
+  /** Return the  spark home version */
+  def SPARK_HOME_VER(version: String): Option[String] = {
```

Comment:
I was thinking more along the lines of adding
`def SPARK_HOME_VER(version: String) = Entry(s"livy.server.spark-home-$version", null)`
to the LivyConf object above as a `def` similar to the `val`'s, then doing `Option(get(SPARK_HOME_VER(version)))` which you could then do an orElse throw Exception on inside the sparkHome def as before.